src/libutil/url.hh

```diff
@@ -10,7 +10,7 @@ struct ParsedURL
     std::string base; // URL without query/fragment
     std::string scheme;
     std::optional<std::string> authority;
-    std::string path;
+    std::string path; // Percent encoded
```

Comment:
Wouldn't it be better to decode `path` in `parseURL()`? That way consumers of `path` don't have to remember to decode it, and avoids having to percent-encode in `parseFlakeRefWithFragment()`.

Reply:
That is incredibly reasonable. Added it.